runtime-sdk/src/dispatcher.rs

```diff
@@ -196,25 +219,33 @@ impl<R: Runtime> Dispatcher<R> {
             return (e.into_call_result(), call_format_metadata);
         }
 
+        // Make sure that a read-only call did not result in any modifications.
+        if call.read_only && ctx.runtime_state().has_pending_updates() {
```

Comment:
If you do this, then it's harder to reuse the normal `Call` method for queries. In my ideal paratime, there are no queries and people can submit read-only txs to any method they can, and get whatever results it produces–just with the caveat that it's not persisted.

From another perspective, is there really any security risk to call an explicitly allowlisted interactive call that _may_ modify state, but then revert it? I guess the runtime itself could revert its own state, but that's a little annoying.